### DISCUSSION TITLE
Bitmap Font Scaling

### DISCUSSION THREAD

**Discussion by MicrotonalMatt** [godotengine/godot-proposals#6150] [GUI]
I've been trying to come up with a workflow for using bitmap fonts with scaling and I can't seem to come up with a solution. This issue, https://github.com/godotengine/godot-proposals/issues/2447 , claims this is implemented in the new Complex Texts Layouts but setting font size still doesn't scale the font as far as I can tell. Unless I'm missing something, I couldn't find a text scale option anywhere and setting font size does nothing for bitmap fonts. Using bitmap fonts in RichTextLabels in containers results in fixed size, since the label scaling is locked to the container. 

I am trying to use bitmap font scaling as a visual reference to how fonts were used on the Nintendo 64 specifically, where fonts were packed into a very small image and then scaled up for use. 

I was advised on the discord to make the label a child of a basic control and scale the label from there, resulting in needing to write a script to resize the label bounds properly afterwards, which is cumbersome for iteration. Using this method in complex menus where all text is presented this way is cluttering up the node tree. The only other option I could come up with was to create images of all the text needed and place them into TextureRects instead, which even clunkier for the volume of text I need.

I think having a font scaling x,y option in font sizes would fix this issue, and if scaling could be used regardless of font type, it might actually be useful for this style of scaling without actually needing to use bitmap fonts, which are already a bit more cumbersome than other font types.

**Comment by Calinou** [comment]
Bitmap fonts ignoring the Font Size theme item is likely a bug.

cc @bruvzg

**Comment by akien-mga** [comment]
Implemented by https://github.com/godotengine/godot/pull/80605.
